src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -630,7 +630,7 @@ var ReactCompositeComponentMixin = {
     if (this._pendingElement != null) {
       ReactReconciler.receiveComponent(
         this,
-        this._pendingElement || this._currentElement,
+        this._pendingElement,
```

Comment:
Sorry, it is not obvious to me why this code is dead.  Can you elaborate?

It is possible that the component rendered a `<div />` last cycle, and is rendering `null` this cycle, in which case I think `_pendingElement ` would be null and `_currentElement` would not be null.